packages/rest/src/rest-application.ts

```diff
@@ -0,0 +1,39 @@
+import {Application, ApplicationConfig, Server} from '@loopback/core';
+import {RestComponent} from './index';
+import {Binding, Constructor} from '@loopback/context';
+import {format} from 'util';
+
+export const ERR_NO_MULTI_SERVER = format(
+  'RestApplication does not support multiple servers!',
+  'To create your own server bindings, please extend the Application class.',
+);
+/**
+ * An implementation of the Application class that automatically provides
+ * an instance of a REST server. This application class is intended to be
+ * a single-server implementation. Any attempt to bind additional servers
+ * will throw an error.
+ *
+ */
+export class RestApplication extends Application {
+  constructor(config?: ApplicationConfig) {
+    const cfg = Object.assign({}, config);
+
+    // If the configuration provides an array of components, we want
+    // to preserve that collection, but also ensure that it contains
+    // RestComponent.
+    if (!cfg.components) {
+      cfg.components = [];
+    }
+    if (!cfg.components.includes(RestComponent)) {
+      cfg.components.push(RestComponent);
+    }
+    super(cfg);
+  }
+
+  server(server: Constructor<Server>, name?: string): Binding {
+    if (this.findByTag('server').length > 0) {
```

Comment:
Shouldn't it be `this.findByTag('server').length > 1` to account for the server that comes with RestApplication?

Reply:
The purpose of the tests is to demonstrate that all of those approaches are actually using the `server` function under the hood, which also includes the automatic instantiation and mounting of the RestServer as a part of RestComponent's loading. 

We want this function to work when no servers are present *because* this function is exactly how we get our first instance. If the count check was > 1, it would mean that we could always get two servers (since 1 > 1 is false, it won't throw)